crates/ruma-common/src/identifiers/crypto_algorithms.rs

```diff
@@ -59,6 +59,40 @@ pub enum EventEncryptionAlgorithm {
     _Custom(PrivOwnedStr),
 }
 
+/// A key algorithm to be used to generate a key from a passphrase.
+///
+/// This type can hold an arbitrary string. To check for algorithms that are not available as a
+/// documented variant here, use its string representation, obtained through `.as_str()`.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, StringEnum)]
+#[non_exhaustive]
+#[cfg_attr(feature = "serde", derive(DeserializeFromCowStr, SerializeAsRefStr))]
+pub enum KeyDerivationAlgorithm {
+    /// PBKDF2
+    #[ruma_enum(rename = "m.pbkdf2")]
+    Pbkfd2,
+
+    #[doc(hidden)]
+    _Custom(PrivOwnedStr),
+}
+
+// /// An encryption algorithm to be used to encrypt a secret to be stored in a user's account_data.
+// ///
+// /// This type can hold an arbitrary string. To check for algorithms that are not available as a
+// /// documented variant here, use its string representation, obtained through `.as_str()`.
+// #[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, StringEnum)]
+// #[non_exhaustive]
+// #[cfg_attr(feature = "serde", derive(DeserializeFromCowStr, SerializeAsRefStr))]
+// pub enum SecretEncryptionAlgorithm {
+//     /// m.secret_storage.v1.aes-hmac-sha2 secret encryption algorithm.
+//     ///
+//     /// The secret is encrypted using AES-CTR-256 and MACed using HMAC-SHA-256.
+//     #[ruma_enum(rename = "m.secret_storage.v1.aes-hmac-sha2")]
+//     SecretStorageV1AesHmacSha2,
+
+//     #[doc(hidden)]
+//     _Custom(PrivOwnedStr),
+// }
```

Comment:
Can this be removed, or do you expect we'll need it at some point?

Reply:
Seems I forgot to commit removing this, I don't think it'll be needed. It's only used in the context of storing key descriptions and secrets, and in those cases we already have separate enums to represent `m.secret_storage.v1.aes-hmac-sha2`.